app/controllers/spree/admin/reports_controller_decorator.rb

```diff
@@ -222,7 +222,8 @@ def my_suppliers
   end
 
   def suppliers_of_products_distributed_by(distributors)
-    distributors.map { |d| Spree::Product.in_distributor(d) }.flatten.map(&:supplier).uniq
+    distributors.map { |d| Spree::Product.in_distributor(d).includes(:supplier).all }.
```

Comment:
Definitely options for improvement here, but I agree it's out of scope. We probably want something like `Enterprise.is_primary_producer.supplying_variant_in(...)`.